### DISCUSSION TITLE
Add-PnPFile -Path <blob storage path>

### DISCUSSION THREAD

**Discussion by phuggins** [pnp/powershell#1775] [Q&A] [answered]
Is there a way to pass a blob storage URI into the -Path parameter for the Add-PnPFile command? I have working code that will read from a local path but would like to connect it to blob storage. I have setup the AzStorageContext and verified that the file exists within the script. When I pass the blob uri into -Path I get a 'Local File was not Found' message. Am I missing something or is this not supported at the moment?

**Comment by AndersRask** [comment, marked as answer]
I usually just save the files from the storage account to a temp dir that I clean up afterwards.
This approach works for Runbooks as well

**Reply by phuggins** [reply]
Great idea, I'm working in a runbook so I'll give this a shot. Thanks for the suggestion!
